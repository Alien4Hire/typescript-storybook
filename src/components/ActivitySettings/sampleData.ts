import { ActivitySettingT } from '../../app/reducers/activitySettings/activitySettingsSlice';

export const fakeData: ActivitySettingT[] = [
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afaa',
    name: 'Setting One',
    recommendedUse: 'none',
    studentVisibility: true,
    templateId: '3fa85f64-5717-4562-b3fc-2c963f66afa6',
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afab',
    name: 'Setting Two',
    recommendedUse: 'none',
    studentVisibility: true,
    templateId: '3fa85f64-5717-4562-b3fc-2c963f66afa6',
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afac',
    name: 'Setting Three',
    recommendedUse: 'none',
    studentVisibility: true,
    templateId: '3fa85f64-5717-4562-b3fc-2c963f66afa6',
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afad',
    name: 'Setting Four',
    recommendedUse: 'none',
    studentVisibility: true,
    templateId: '3fa85f64-5717-4562-b3fc-2c963f66afa6',
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afae',
    name: 'Setting Five',
    recommendedUse: 'pre-class',
    studentVisibility: true,
    templateId: '3fa85f64-5717-4562-b3fc-2c963f66afa6',
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afaf',
    name: 'Setting Six',
    recommendedUse: 'pre-class',
    studentVisibility: true,
    templateId: '3fa85f64-5717-4562-b3fc-2c963f66afa6',
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afag',
    name: 'Setting Seven',
    recommendedUse: 'pre-class',
    studentVisibility: true,
    templateId: '3fa85f64-5717-4562-b3fc-2c963f66afa6',
  },
];
