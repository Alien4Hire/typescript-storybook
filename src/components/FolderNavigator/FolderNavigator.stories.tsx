import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import FolderNavigator from './FolderNavigator';
import { store } from '../../app/store';

export default {
  title: 'Component/FolderNavigator',
  component: FolderNavigator,
} as ComponentMeta<typeof FolderNavigator>;

const Template: ComponentStory<typeof FolderNavigator> = () => (
  <Provider store={store}>
    <Router>
      <FolderNavigator />
    </Router>
  </Provider>
);

export const Primary = Template.bind({});
Primary.args = {};
