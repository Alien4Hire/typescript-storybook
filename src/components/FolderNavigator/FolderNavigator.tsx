import React from 'react';
import { Box } from '@mui/material';
import RecursiveTreeView from '../RecursiveTreeView/RecursiveTreeView';

const FolderNavigator = () => {
  return (
    <Box>
      <RecursiveTreeView />
    </Box>
  );
};

export default FolderNavigator;
