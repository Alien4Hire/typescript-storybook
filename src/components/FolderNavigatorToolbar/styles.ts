import { Box, Button } from '@mui/material';

import { styled } from '@mui/material/styles';

export const Divider = styled('div')(({ theme }) => ({
  borderRight: `1px solid ${theme.palette.gray.medium}`,
  height: '34px',
  margin: 'auto',
}));

export const Label = styled('div')(({ theme }) => ({
  fontWeight: '600',
  fontSize: '15px',
  letterSpacing: '0.4px',
  color: theme.palette.gray.veryDark,
  fontFamily: "'Source Sans Pro', sans-serif",
}));

export const Wrapper = styled('div')(({ theme }) => ({
  position: 'relative',
  background: theme.palette.gray.veryLight,
  paddingLeft: '56px',
}));

export const FolderForm = styled('div')`
  display: flex;
  align-items: center;
`;

export const FolderAdd = styled(Box, {
  shouldForwardProp: (prop: string) => prop !== 'active',
})(({ active, theme }) => ({
  position: 'absolute',
  top: 0,
  bottom: 0,
  left: 0,
  right: 0,
  width: 'auto',
  padding: '2px',
  backgroundColor: theme.palette?.gray.veryLight,
  display: 'none',
  ...(active && {
    display: 'flex',
    alignItems: 'center',
  }),
}));

export const CsvModalBackdrop = styled(Box)({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: '#f1f1f1',
  borderRadius: '16px',
  padding: '32px 32px 70px',
});

export const UploadListItem = styled(Box)(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  padding: '16px 0px',
  borderBottom: `1px solid ${theme.palette.gray.light}`,
}));

export const UploadButton = styled(Button)({
  textTransform: 'none',
  width: 'fit-content',
  marginTop: '16px',
});

export const DownloadTemplateWrapper = styled(Box)({
  display: 'flex',
  alignItems: 'center',
  marginTop: '24px',
  cursor: 'pointer',
});
