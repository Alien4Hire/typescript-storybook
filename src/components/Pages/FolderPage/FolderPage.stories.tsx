import { MemoryRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { store } from '../../../app/store';
import FolderPage from './index';

export default {
  title: 'Pages/FolderPage',
  component: FolderPage,
} as ComponentMeta<typeof FolderPage>;

const Template: ComponentStory<typeof FolderPage> = (args) => (
  <Provider store={store}>
    <Router>
      <FolderPage />
    </Router>
  </Provider>
);

export const Primary = Template.bind({});
