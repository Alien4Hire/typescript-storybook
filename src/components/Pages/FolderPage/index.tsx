import { CircularProgress } from '@mui/material';
import { useAppSelector, useAppDispatch } from '../../../app/hooks';
import { getFilterStatus } from '../../../app/reducers/resourcesFilter/resourcesFilterSlice';
import { useGetTemplateViewTypesQuery } from '../../../app/reducers/templateViewTypes/templateViewTypesAPI';
import FilterResources from '../../FilterResources';
import FolderNavigator from '../../FolderNavigator/FolderNavigator';
import Toolbar from '../../Toolbar/Toolbar';
import TargetFolderSideBar from '../../../components/TargetFolderSideBar';
import { getLessons } from '../../../app/reducers/lessons/lessonsSlice';
import {
  getTargetSideBarStatus,
  resetTargetFolderSideBarState,
} from '../../../app/reducers/targetFolderSideBar/targetFolderSideBarSlice';
import { Box } from '@mui/material';
import { useEffect } from 'react';

const FolderPage = () => {
  const dispatch = useAppDispatch();
  const isFilterOpen: boolean = useAppSelector(getFilterStatus);
  const isSideBarOpen = useAppSelector(getTargetSideBarStatus);
  const { items: folderList } = useAppSelector(getLessons);

  useEffect(() => {
    return () => {
      dispatch(resetTargetFolderSideBarState());
    };
  }, [dispatch]);
  const { isSuccess: templateViewTypesFetchSuccess } =
    useGetTemplateViewTypesQuery(undefined);

  // Ran into edge-case race condition where view types not
  // yet ready. This checks for successful fetch before
  // continuing.
  if (!templateViewTypesFetchSuccess) {
    return <CircularProgress />;
  }

  return (
    <>
      <Box
        sx={{
          display: 'flex',
          flexDirection: 'column',
        }}
      >
        <Box sx={{ marginBottom: '10px' }}>
          <Toolbar color="primary" />
        </Box>
        <Box
          sx={{
            display: 'flex',
            flexGrow: 1,
            minHeight: '100%',
          }}
        >
          <Box sx={{ width: '100vw' }}>
            {' '}
            {isFilterOpen ? <FilterResources /> : <FolderNavigator />}
          </Box>
          <Box>
            <TargetFolderSideBar
              folders={folderList?.children}
              isOpen={isSideBarOpen}
            />
          </Box>
        </Box>
      </Box>
    </>
  );
};

export default FolderPage;
