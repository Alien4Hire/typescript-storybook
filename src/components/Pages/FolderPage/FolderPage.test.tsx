import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';

import { children } from '../../FolderNavigator/sampleData';
import FolderPage from './index';
import { store } from '../../../app/store';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';
import { theme } from '../../../theme';
import { useGetTemplateViewTypesQuery } from '../../../app/reducers/templateViewTypes/templateViewTypesAPI';
jest.mock('../../../app/reducers/templateViewTypes/templateViewTypesAPI');
const useGetTemplateViewTypesQueryMock =
  useGetTemplateViewTypesQuery as jest.Mock;

beforeEach(() => {
  fetchMock.resetMocks();
  jest.clearAllMocks();
  fetchMock.mockResponse(JSON.stringify(children));
});

const types = [
  {
    id: 'd4fc4e46-b9e1-45ba-9a16-1c665fd8ce64',
    name: 'Resource Structure',
    description: 'Base list of activities for course',
  },
  {
    id: 'd4fc4e46-b9e1-45ba-9a16-1c665fd8ce65',
    name: 'Prebuilt Course',
    description: 'Base list of activities for course',
  },
];

beforeEach((): void => {
  jest.clearAllMocks();
});
const renderComponent = () => {
  return render(
    <Provider store={store}>
      <Router>
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={theme}>
            <FolderPage />
          </ThemeProvider>
        </StyledEngineProvider>
      </Router>
    </Provider>
  );
};

describe('<FolderPage />', () => {
  test('renders without crashing', () => {
    useGetTemplateViewTypesQueryMock.mockReturnValue({ data: types });
    const view = renderComponent();
    expect(view).toMatchSnapshot();
  });
});
