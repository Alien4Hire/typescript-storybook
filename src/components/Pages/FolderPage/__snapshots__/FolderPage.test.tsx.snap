// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FolderPage /> renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: inline-block;
  color: #00758e;
  -webkit-animation: epmXAj 1.4s linear infinite;
  animation: epmXAj 1.4s linear infinite;
}

.c1 {
  display: block;
}

.c2 {
  stroke: currentColor;
  stroke-dasharray: 80px,200px;
  stroke-dashoffset: 0;
  -webkit-animation: jzfMP 1.4s ease-in-out infinite;
  animation: jzfMP 1.4s ease-in-out infinite;
}

<body>
    <div>
      <span
        class="c0 MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="c1 MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="c2 MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </body>,
  "container": .c0 {
  display: inline-block;
  color: #00758e;
  -webkit-animation: epmXAj 1.4s linear infinite;
  animation: epmXAj 1.4s linear infinite;
}

.c1 {
  display: block;
}

.c2 {
  stroke: currentColor;
  stroke-dasharray: 80px,200px;
  stroke-dashoffset: 0;
  -webkit-animation: jzfMP 1.4s ease-in-out infinite;
  animation: jzfMP 1.4s ease-in-out infinite;
}

<div>
    <span
      class="c0 MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="c1 MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="c2 MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
